packages/core/src/RenderPipeline/BasicRenderPipeline.ts

```diff
@@ -169,6 +169,7 @@ export class BasicRenderPipeline {
       rhi.clearRenderTarget(camera.engine, clearFlags, color);
     }
 
+    scene._maskManager.preMaskLayer = 0;
```

Comment:
If the stencil buffer is not cleared, does this value need to be reset? 
If the `renderTarget` is changed, does this value need to be reset?

Reply:
Must cleard when use a camera to render, stencil just support compare with one reference value, unless not use mask diff

Reply:
`preMaskLayer` must be a property of the `RenderTarget` object:
- `clearStencil` will directly affect the value of `preMaskLayer`
- When `MaskManager.addSpriteMask` or `MaskManager.removeSpriteMask` is executed, `RenderTarget` will be notified to do diff based on `preMaskLayer`